pkg/security/reporter/reporter.go

```diff
@@ -52,7 +52,7 @@ func newReporter(hostname string, stopper startstop.Stopper, sourceName, sourceT
 	stopper.Add(auditor)
 
 	// setup the pipeline provider that provides pairs of processor and sender
-	pipelineProvider := pipeline.NewProvider(logsconfig.NumberOfPipelines, auditor, &diagnostic.NoopMessageReceiver{}, nil, endpoints, context, agentimpl.NewStatusProvider(), hostnameimpl.NewHostnameService(), pkgconfigsetup.Datadog())
+	pipelineProvider := pipeline.NewProvider(pkgconfigsetup.Datadog().GetInt("logs_config.pipelines"), auditor, &diagnostic.NoopMessageReceiver{}, nil, endpoints, context, agentimpl.NewStatusProvider(), hostnameimpl.NewHostnameService(), pkgconfigsetup.Datadog())
```

Comment:
This code path can be called from the system probe, and should probably take the config itself as an input, is there a default you can use in this PR and we can take care of making it configurable in a later PR.

I'm also not a huge fan of having a `logs_config` option configuring some CWS related code.

Reply:
related to https://github.com/DataDog/datadog-agent/pull/31190#issuecomment-2483934221, I think a default would be best for now (either 4 or rutime.GOMAXPROCS)

Reply:
Thanks for the feedback - I defaulted it to `GOMAXPROCS` and avoided the logs agent config.

Reply:
I am not super aware of your use case. But FWIW - in the future, unless you are sending an extremely high volume (10+ MB/sec) of data you probably don't need more than 1 pipeline which could same some memory overhead.

Reply:
Yeah we are definitely not sending >10MB/sec, IMO your default is good for now, and you can do the same in `pkg/compliance`. I will do some benchmarks/measurements once this is merged to see if 1 is better. Thanks a lot !